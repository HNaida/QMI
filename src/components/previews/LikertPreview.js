import React from "react";
import {Grid, Radio, Typography} from "@material-ui/core";

const LikertPreview = ({question}) => {
    const renderBox = () =>
        <Grid
            item
            xs={"auto"}
            style={{
                textAlign: "center",
            }}
        >
            <Radio disabled/>
        </Grid>;

    const printOption = (option) => {
        return option.numeric_value === undefined ? option : option.title + " (" + option.numeric_value + ")"
    };

    const renderText = (option) =>
        <Grid
            item
            xs={"auto"}
            style={{
                wordWrap: "break-word",
                textAlign: "left",
            }}
        >
            <Typography variant="body1">{printOption(option)}</Typography>
        </Grid>;


    const renderOption = (option) =>
        <Grid
            item
            xs={"auto"}
            key={option.numeric_value}
        >
            <Grid
                container
                direction="column"
                justify="flex-start"
                alignItems="center"
            >
                {renderBox()}
                {renderText(option)}
            </Grid>
        </Grid>;

    return (
        <Grid
            container
            direction="row"
            justify="space-evenly"
            spacing={1}
        >
            {question.options.map(option => (renderOption(typeof (option) === "string" ? option : option.title)))}
        </Grid>
    );
};

export default LikertPreview;
