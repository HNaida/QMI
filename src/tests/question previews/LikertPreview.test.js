import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react'
import LikertPreview from '../../components/previews/LikertPreview'

configure({ adapter: new Adapter() });

describe('<LikertPreview/>', () => {
    it('renders a likert question', () => {
        const question= {
            id: "v4",
            type: "likert",
            title: "untitled likert",
            options: [
                {title: "hobby/sport"},
                {title: "werk"},
                {title: "vriendschap"},
            ],
        };
        const component = shallow(<LikertPreview question={question}/>);
        // expect(component).toMatchSnapshot();
    });
});
