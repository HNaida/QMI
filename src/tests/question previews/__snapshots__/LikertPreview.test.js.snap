// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikertPreview/> renders a number question 1`] = `ShallowWrapper {}`;
